examples/MLIRVector/vector-broadcast.mlir

```diff
@@ -17,6 +19,117 @@ func.func @main() -> i32 {
   %broadcast_vec_2d = vector.broadcast %load_vec : vector<4xf32> to vector<4x4xf32>
   vector.print %broadcast_vec_2d : vector<4x4xf32>
 
+  // Detailed example for all cases in broadcast:
+  %v1 = arith.constant dense<[[1.0], [2.0], [3.0]]> : vector<3x1xf32>
+  %v2 = arith.constant dense<[[[1.0], [2.0], [3.0]]]> : vector<1x3x1xf32>
+
+  func.call @broadcast_scalar_to_vector(%f1) : (f32) -> vector<3x1xf32>
+  func.call @broadcast_low_dim_to_high_dim(%v1) : (vector<3x1xf32>) -> vector<4x3x2xf32>
+  func.call @broadcast_1_to_n_case(%v2) : (vector<1x3x1xf32>) -> vector<4x3x2xf32>
+  func.call @broadcast_n_to_n_case(%v1) : (vector<3x1xf32>) -> vector<3x2xf32>
+
   %ret = arith.constant 0 : i32
   return %ret : i32
 }
+
+func.func @broadcast_scalar_to_vector(%src: f32) -> vector<3x1xf32> {
+  // broadcast a scalar to vector
+  %result = vector.broadcast %src : f32 to vector<3x1xf32>
+
+  // equals to splat it, or, "fill" it
+  // like [[%src], [%src], [%src]]
+  %w = vector.splat %src : vector<3x1xf32>
+
+  // %w == %result
+  vector.print %result : vector<3x1xf32>
+  vector.print %w : vector<3x1xf32>
+
+  return %result : vector<3x1xf32>
+}
+
+func.func @broadcast_low_dim_to_high_dim(%src: vector<3x1xf32>) -> vector<4x3x2xf32> {
+  // broadcast vector from low dimension to high can be transform into 
```

Comment:
This is not completely clear. Can you please rephrase it.

Reply:
The sentence acts as a summary for this case, and the behavior it summarizes is described in detail below. Can you please give a little more detail about why you think it's unclear?

This sentence has had minor revisions to improve clarity in latest commit.